### DISCUSSION TITLE
Will logto alter database schema automatically in upgrading ?

### DISCUSSION THREAD

**Discussion by zhfkt** [logto-io/logto#5637] [Q&A]
I am reading https://docs.logto.io/docs/references/using-cli/database-alteration/ . Will logto alter database schema automatically in upgrading ? Or I need to do it manually ?

For example, if I upgrade from `v1.13.0` to `v1.14.0`, will I need to run `logto db alteration deploy` manually ?

**Comment by xiaoyijun** [comment]
Yes, you need to run that command manually.
However, I'm curious about when would you need it to execute automatically?

**Reply by zhfkt** [reply]
Usually database migration is executed automatically in upgrading such as `flyway` (https://flywaydb.org/).  Manual execution is rare. It is easier for user to upgrade application if database alteration is executed automatically  on **startup** instead of user executing cli `logto db alteration deploy`.

![image](https://github.com/logto-io/logto/assets/4969804/f520ade2-c3d3-4c2a-bceb-9ebd513115d3)

**Reply by xiaoyijun** [reply]
In fact, you can write the alteration command into your startup command, so it can be executed automatically at startup. Because operations on the database need to be cautious, Logto does not perform alteration automatically at startup, but leaves it to users to decide the timing of alteration.

**Reply by zhfkt** [reply]
Thanks. Pls also consider automatically altering it since `Keycloak` was migrating database currently.
